util/topgen/templates/toplevel.sv.tpl

```diff
@@ -37,6 +37,8 @@ for m in top['memory']:
   if m['type'] == 'rom':
     has_toplevel_rom = True
 
+last_module_with_params = lib.idx_of_last_module_with_params(top)
```

Comment:
maybe factor `idx` into the name to indicate its just an integer and not a full module.